src/main/java/org/carlmontrobotics/robotcode2023/commands/AlignChargingStation.java

```diff
@@ -0,0 +1,64 @@
+package org.carlmontrobotics.robotcode2023.commands;
+
+import static org.carlmontrobotics.robotcode2023.Constants.Drivetrain.*;
+
+import org.carlmontrobotics.robotcode2023.subsystems.Drivetrain;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class AlignChargingStation extends CommandBase {
+
+    private final Drivetrain drivetrain;
+    private double lastTime = -1;
+    private boolean fwd, fieldOriented;
+
+    public AlignChargingStation(Drivetrain drivetrain) {
+        addRequirements(this.drivetrain = drivetrain);
+    }
+
+    @Override
+    public void initialize() {
+        fwd = Math.abs(getPitch()) > Math.abs(getRoll());
+        fieldOriented = drivetrain.getFieldOriented();
+        drivetrain.setFieldOriented(false);
+    }
+
+    @Override
+    public void execute() {
+        double pitch = getPitch();
+        double roll = getRoll();
+        double forward = fwd && Math.abs(pitch) > chargeStationAlignTolerance ? chargeStationAlignSpeed * pitch + Math.copySign(chargeStationAlignFF, pitch) : 0;
+        double strafe = !fwd && Math.abs(roll) > chargeStationAlignTolerance ? chargeStationAlignSpeed * roll + Math.copySign(chargeStationAlignFF, roll) : 0;
```

Comment:
This might "work" but at a minimum `chargeStationAlignSpeed` and `chargeStationAlignFF` should be renamed because they don't have their normal units of m/s and volts respectively.

Even better, use the drivetrain's `SimpleFeedForward` model to compute the voltage required to overcome the tilt and move at the desired velocity.

Reply:
I've updated the variable names to include units. As for using a Feedforward model, atm I don't think we have FF constants for the drivetrain as a whole, just the individual wheels. Also, those don't account for gravity, instead assuming that it will always be canceled out by the normal force.

Reply:
Ah, yes I remember proposing that in the initial versions of the command in addition to what we have now. Can't recall why we ended up with this solution. We should maybe consider it for future revisions, but atm I'm inclined to keep it as-is.